aws-blog-event-driven-batch-analytics/src/main/java/com/amazonaws/bigdatablog/edba/LambdaContainer.java

```diff
@@ -0,0 +1,249 @@
+		Connection conn = DriverManager.getConnection(url, user, password);
+		ResultSet openStepsRS = conn.createStatement().executeQuery(retrieveOpenSteps);
+		AmazonElasticMapReduceClient emr = new AmazonElasticMapReduceClient();
+		DescribeStepRequest stepReq=new  DescribeStepRequest();
+		PreparedStatement ps = conn.prepareStatement(updateStepStatus);
+		while(openStepsRS.next()){
+			
+			stepReq.setClusterId(openStepsRS.getString("cluster_id"));
+			stepReq.setStepId(openStepsRS.getString("step_id"));
+			String stepState = emr.describeStep(stepReq).getStep().getStatus().getState();
+			
+				if(stepState.equals(StepState.COMPLETED.toString())){
+					ps.setString(1,StepState.COMPLETED.toString());
+				}else if (stepState.equals(StepState.FAILED.toString())){
+					ps.setString(1,StepState.FAILED.toString());					
+				}
+				ps.setString(2,openStepsRS.getString("job_config_id"));
+				ps.addBatch();
+		}
+		
+		ps.executeBatch();
+		ps.close();
+		conn.close();
+	}
+	
+	// adds EMR step
+	protected String fireEMRJob(String paramsStr,String clusterId){
+		StepFactory stepFactory = new StepFactory();
+		AmazonElasticMapReduceClient emr = new AmazonElasticMapReduceClient();
+		emr.setRegion(Region.getRegion(Regions.US_EAST_1));
+		Application sparkConfig = new Application()
+				.withName("Spark");
+		
+		String[] params = paramsStr.split(",");
+		StepConfig enabledebugging = new StepConfig()
+				.withName("Enable debugging")
+				.withActionOnFailure("TERMINATE_JOB_FLOW")
+				.withHadoopJarStep(stepFactory.newEnableDebuggingStep());
+		
+		HadoopJarStepConfig sparkStepConf = new HadoopJarStepConfig()
+				.withJar("command-runner.jar")
+				.withArgs(params);	
+		
+		final StepConfig sparkStep = new StepConfig()
+				.withName("Spark Step")
+				.withActionOnFailure("CONTINUE")
+				.withHadoopJarStep(sparkStepConf);
+
+		
+		AddJobFlowStepsRequest request = new AddJobFlowStepsRequest(clusterId)
+				.withSteps(new ArrayList<StepConfig>(){{add(sparkStep);}});
+				
+
+		AddJobFlowStepsResult result = emr.addJobFlowSteps(request);
+		return result.getStepIds().get(0);
+	}
+	
+	protected List<String> getActiveClusterId() throws Exception{
+		AmazonElasticMapReduceClient emrClient = new AmazonElasticMapReduceClient();
+		List<String> waitingClusters = new ArrayList<String>();
+		ListClustersResult clusterResult = emrClient.listClusters(new ListClustersRequest().withClusterStates(ClusterState.WAITING));
+		for( ClusterSummary cluster : clusterResult.getClusters()){
+				waitingClusters.add(cluster.getId());
+		}
+		return waitingClusters;
+		
+	}
+	/**
+	 * Checks whether additional criteria returned a non empty resultset.
+	 */
+	protected boolean isAdditionalCriteriaPassed(String sql, Connection conn) throws Exception{
+		if(StringUtils.isNullOrEmpty(sql)){
+			return true;
+		}
+		ResultSet rs = conn.createStatement().executeQuery(sql);
+		if(!rs.next()){
+			return false; // Empty Resultset
+		}
+		return true;
+	}
+	
+
+	static{
```

Comment:
Created the Driver instance as advised. Removed loading the jdbc drivers from class loader